File: single-language-analysis/dataset-files/java/TestSmell107.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell107 {

    @Test
    public void testLfu() {
        String storage = "file"; // Change this to "memory" for the other case
        String filePath = storage.equals("memory") ? null : "tmpdir/cache";
        Cache cache = new Cache(filePath, 2, -1, "LFU");

        Function<Integer, Integer> func = a -> a;

        assertEquals(1, func.apply(1).intValue());
        assertEquals(2, func.apply(2).intValue());
        List<Integer> theKeys = cache.items().stream()
            .map(entry -> entry.getKey().getSecond())
            .collect(Collectors.toList());
        assertEquals(2, theKeys.size());
        assertTrue(theKeys.contains(1));
        assertTrue(theKeys.contains(2));

        assertEquals(1, func.apply(1).intValue());
        assertEquals(1, func.apply(1).intValue());
        assertEquals(2, func.apply(2).intValue());
        assertEquals(2, func.apply(2).intValue());
        assertEquals(3, func.apply(3).intValue());
        theKeys = cache.items().stream()
            .map(entry -> entry.getKey().getSecond())
            .collect(Collectors.toList());
        assertEquals(2, theKeys.size());
        assertTrue(theKeys.contains(1));
        assertTrue(theKeys.contains(2));
        assertFalse(theKeys.contains(3));
    }
}

File: single-language-analysis/dataset-files/python/test_smell_107.py
# Original URL: https://github.com/bofm/python-caching/blob/16de50c119f8896e923e3f10488ecf6ffe535f45/tests/test_cache.py#L301-L327

@pytest.mark.parametrize('storage', ['file', 'memory'])
def test_smell_107(tmpdir, storage):
    filepath = None if storage == 'memory' else f'{tmpdir}/cache'
    cache = Cache(filepath=filepath, maxsize=2, ttl=-1, policy='LFU')

    @cache
    def func(a):
        return a

    def keys():
        return [arg for (fn_name, arg), v in cache.items()]

    assert func(1) == 1
    assert func(2) == 2
    the_keys = keys()
    assert len(the_keys) == 2
    assert 1 in the_keys and 2 in the_keys

    assert func(1) == 1
    assert func(1) == 1
    assert func(2) == 2
    assert func(2) == 2
    assert func(3) == 3
    the_keys = keys()
    assert len(the_keys) == 2
    assert 1 in the_keys and 2 in the_keys
    assert 3 not in the_keys